### PR TITLE
Update some known DG bugs reported by Pe-D

### PR DESCRIPTION
**Pull request by Jianwen0451**
#230 
1 Update the glossary so it is consistent with the archive and load.
#239 #214 
2 Write some test cases in the Appendix to guide the tester.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -675,7 +675,36 @@ testers are expected to do more *exploratory* testing.
    1. Re-launch the app by double-clicking the jar file.<br>
        Expected: The most recent window size and location is retained.
 
-1. _{ more test cases …​ }_
+1. Exit the App
+    1. Enter `exit` in the command box. This will exit the app.
+
+### Adding a person
+1. Adding a person into the person list
+    1. Assumption: We assume there is no duplicate in this test case
```
Comment (yongqqqq) [inline docs/DeveloperGuide.md:683]: Indentation for pointer 1. can be standardised

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -692,12 +721,65 @@ testers are expected to do more *exploratory* testing.
...
 
 ### Saving data
 
 1. Dealing with missing/corrupted data files
-
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
-
-1. _{ more test cases …​ }_
+    1. Here `<HomeFolder>` refer to the folder containing the jar file.
+   1. Test case: Missing data file<br>
```
Comment (yongqqqq) [inline docs/DeveloperGuide.md:730]: indentation can be standardized

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -692,12 +721,65 @@ testers are expected to do more *exploratory* testing.
...
+   Expected: a sample database will be provided.
+   2. Test case: Corrupted file<br>
+      1. Ensure that there is at least 1 student in StoreClass
+      1. Close the app
+      2. Open the json file with path  `<HomeFolder>/data/addressbook.json` and edit the name of the first student name to a invalid one e.g. `J@hn D*n`
+      3. ReOpen the app <br>
+   Expected: the corrupted list is discarded and an empty list is provided
+
+1. Saving data
+    1. Do some simple command to add / edit / delete student
```
Comment (yongqqqq) [inline docs/DeveloperGuide.md:743]: Indentation can be standardized

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -692,12 +721,65 @@ testers are expected to do more *exploratory* testing.
...
+
+### Filtering a list
+
+1. Test case: `filter g/male`<br>
+Expected: list out all the male students.
+
+### Undoing and Redoing a action
+
+1. Test case: Undo and redo a command
+    1. Perform a simple command that is supported by undo and redo e.g. add, edit or delete
```
Comment (yongqqqq) [inline docs/DeveloperGuide.md:761]: indentation can be standardized

**Review by yongqqqq — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -642,7 +642,7 @@ Priorities: High (must have) - `* * *`, Medium (Good to have) - `* *`, Low (nice
 ### Glossary
 
 * **Archive**: A feature that allows users to store old data for use later without cluttering the current interface.
-* **Export**: Saving the student data in a file format such as `.csv` or `.txt` for external use.
+* **Export**: Saving the student data in a file format as `json` file. This is done through data archiving.
 * **Private contact detail**: A contact detail that is not meant to be shared with others
 * **Mainstream OS**: Windows, Linux, Unix, MacOS
```
Comment (tanyqcollin) [inline docs/DeveloperGuide.md:647]: I think should add full stop on these two lines to make it more consistent in this glossary.

**Review by tanyqcollin — APPROVED** [review]

**Pull request merged**